Modules/_sqlite/connection.c

```diff
@@ -279,11 +279,18 @@ pysqlite_connection_init_impl(pysqlite_Connection *self,
     self->NotSupportedError     = state->NotSupportedError;
 
     if (PySys_Audit("sqlite3.connect/handle", "O", self) < 0) {
-        return -1;
+        return -1;  // Don't goto error; at this point, dealloc will clean up.
     }
 
     self->initialized = 1;
     return 0;
+
+error:
+    // There are no statements or other SQLite objects attached to the
+    // database, so sqlite3_close() should always return SQLITE_OK.
+    rc = sqlite3_close(db);
+    assert(rc == SQLITE_OK), rc;
```

Comment:
```suggestion
    assert(rc == SQLITE_OK);
```
This isn't Python :)

Reply:
Oh, this is to avoid warnings for non-debug builds ;)

So for a debug build, the assert will use `rc`. For a non-debug build, the assert is a no-op, so `rc` is left unused; you _may_ get a compiler warning saying that the result variable `rc` is now unused. This trick is also used other places in the code base.

Reply:
Oh I see, thanks! Unfortunate then that there is a GitHub Actions annotation.

Reply:
Yes, GHA annotations help a lot, but they seem to have a slight delay before appearing.

Reply:
GHA annotations still does not like this, since it is de-facto unused. I should probably fix that, since it will appear in other people's PRs.